StackoverflowChatbot/Discord.cs

```diff
@@ -88,6 +89,28 @@ private static Task ClientRecieved(SocketMessage arg)
 			return Task.CompletedTask;
 		}
 
+        private static string BuildSoMessage(SocketGuildUser user, Config.Base config, SocketMessage arg)
+        {
+			var displayname = string.IsNullOrEmpty(user.Nickname) ? user.Username : user.Nickname;
+            var messageStart = $@"\[**[{displayname}]({config.DiscordInviteLink})**]";
+			var messageContent = arg.Content;
+			foreach (var mentionedUser in arg.MentionedUsers)
+			{
+				messageContent.Replace(mentionedUser.Mention, $"@{mentionedUser.Username}");
+			}
+			foreach (var mentionedRoles in arg.MentionedRoles)
+			{
+				messageContent.Replace(mentionedRoles.Mention, $"[@{mentionedRoles.Name}]({config.DiscordInviteLink})");
+			}
+			
```

Comment:
If you can fit in tidying up the mentioned channels too, that'd be ideal. When someone mentions a channel it goes a bit to shit.

Reply:
I'll do that